apps/api/src/campaign/campaign.service.ts

```diff
@@ -140,6 +140,34 @@ export class CampaignService {
     return campaign
   }
 
+  async getCampaignByIdWithDefaultVault(campaignId: string): Promise<Campaign> {
+    const campaign = await this.prisma.campaign.findFirst({
+      where: { id: campaignId },
+      include: {
+        campaignFiles: true,
+        vaults: true,
+        incomingTransfers: { select: { amount: true } },
+      },
+    })
+    if (!campaign) {
+      Logger.warn('No campaign record with ID: ' + campaignId)
+      throw new NotFoundException('No campaign record with ID: ' + campaignId)
+    }
+
+
+    if (campaign.vaults.length == 0) {
+      throw new NotFoundException('No vaults found for campaign: ' + campaignId)
+    }
+
+    for(const vault of campaign.vaults) {
+      campaign['defaultVault'] = vault.id
+      break
+    }
```

Comment:
There is something strange here. If we just want to assign the first vault as default, then no need of a for cycle at all.

Reply:
I agree. I will fix it.